### DISCUSSION TITLE
Search example broken?

### DISCUSSION THREAD

**Discussion by pwfraley** [django-commons/django-unicorn#398] [General]
This looks like a very promising project.  I looked at the examples and stumbled about a problem in the search example.

When you type realy fast, than all is ok, but if you pause just half a second then things go wrong.

Go to the search example and then type (type slowly and finish typing):

Wyoming

The result is georgia and the input only contains the last letter typed in case of Wyoming it is g.

Is this a bug or is this what to expect of django-unicorn?

**Comment by adamghill** [comment]
Unfortunately, I can't replicate what you are seeing. This is what happens for me whether I type fast or slow:

![Kapture 2022-04-14 at 20 47 07](https://user-images.githubusercontent.com/317045/163499750-40026795-803b-45b9-97e8-da0028a966ee.gif)

However, I wouldn't be surprised if there could be lots of improvements in Unicorn to deal with longer response times. One experiment I was trying previously was to force requests into a serial queue to make sure that responses were sent back in the same order they were requested. Honestly, it made the code very complicated and I'm not sure if it was worth it.

However, there are probably other improvements to the debounce function I use in JavaScript or overall Python performance improvements to reduce the request/response time as much as possible.

I'd love any help making the project more useful so any help you can provide would be much appreciated. 👍

**Comment by aidenbell** [comment]
Just FYI, the search example is broken for me on Chrome `102.0.5005.61`. The main issue is that for each update, the prior input into the search box clears. As the update happens fairly quickly (good), if I try and type "tex", the update happens after "te" and then the "x" keypress registers and does a new search for only "x".

At minimum too, using typeahead i'd always expect the input to contain what the user typed after each search result with the cursor/focus in the search input at the correct location.

I think this is an important example and the first friction point I found when looking through the documentation to assess use.

![Peek 2022-05-31 09-45](https://user-images.githubusercontent.com/1640009/171132590-5d9c5039-bba9-4053-b032-d8c90e25fe1c.gif)
